util.py

```diff
@@ -0,0 +1,349 @@
+"""This file defines the utils for the model.
+
+This file contains helper functions used by train_model.py. It defines
+function for creating the vocabulary and loading and preprocessing the 
+dataset. Additionally, it contains functions to compute certain
+metrics like F1 score, sentence-level semantic frame accuracy, etc.
+Some of the code was in this file was taken from 
+https://github.com/ZephyrChenzf/SF-ID-Network-For-NLU/blob/master/utils.py
+
+"""
+
+import os
+import sys
+import tensorflow as tf
+import numpy as np
+import time
+
+def createVocabulary(input_path, output_path, no_pad=False, no_unk=False):
+    if not isinstance(input_path, str):
+        raise TypeError('input_path should be string')
+
+    if not isinstance(output_path, str):
+        raise TypeError('output_path should be string')
+
+    vocab = {}
+    with open(input_path, 'r') as fd, \
+            open(output_path, 'w+') as out:
+        for line in fd:
+            line = line.rstrip('\r\n')
+            words = line.split()
+
+            for w in words:
+                if w == '_UNK':
+                    pass
+                if str.isdigit(w) == True:
+                    w = '0'
+                if w in vocab:
+                    vocab[w] += 1
+                else:
+                    vocab[w] = 1
+        if no_pad == False:
+            vocab = ['_PAD', '_UNK'] + sorted(vocab, key=vocab.get, reverse=True)
```

Comment:
Recommend creating package level constants for  '_PAD', '_UNK'

PADDING_TOKEN = '_PAD'